mockgen/generic_go118.go

```diff
@@ -114,3 +126,5 @@ func (p *fileParser) parseGenericMethod(field *ast.Field, it *namedInterface, if
 
 	return p.parseMethod(nf, it, iface, pkg, tps)
 }
+
+var errIgnore = errors.New("parser decided to ignore the interface for constraints")
```

Comment:
Consider giving this a name that will make it clear what case we are ignoring in `parse.go`. We can also simplify the error message itself.

```suggestion
var errConstraintInterface = errors.New("interface contains constraints")
```